static/src/javascripts/projects/commercial/modules/article-body-adverts.js

```diff
@@ -63,154 +101,81 @@ define([
         var ad = createSlot(name, type);
 
-        function insertion (ad, para) {
+        return fastdom.write(function () {
             para.parentNode.insertBefore(ad, para);
-        }
-
-        // If on mobile we will
-        // insert ad using steady page
-        // to avoid jumping the user
-        if (detect.isBreakpoint({max: 'tablet'})) {
-            return steadyPage.insert(ad, function(){
-                insertion(ad, para);
-            });
-        } else {
-            // If we're not on mobile we insert and resolve the promise immediately
-            return new Promise(function(resolve){
-                insertion(ad, para);
-                resolve();
-            });
-        }
-    }
-
-    function addSlots(countAdded) {
-        if (countAdded > 0) {
-            qwery('.ad-slot--inline').forEach(addSlot);
-        }
-    }
-
-    // If we're on mobile, we want to use steady-page right before dom insertion
-    // when we have the adslot so we provide a non-fastdom writer as
-    // fastdom is handled in steady-page
-    function writerOverride (writerCallback) {
-        return writerCallback();
-    }
-
-    // If a merchandizing component has been rendered but is empty,
-    // we allow a second pass for regular inline ads. This is because of
-    // the decoupling between the spacefinder algorithm and the targeting
-    // in DFP: we can only know if a slot can be removed after we have
-    // received a response from DFP
-    var waitForMerch = memoize(function () {
-        return trackAdRender('dfp-ad--im').then(function (isLoaded) {
-            return isLoaded ? 0 : addArticleAds(2, getRules());
-        }).then(function (countAdded) {
-            return countAdded === 2 ?
-                addArticleAds(8, getLongArticleRules()).then(function (countAdded) {
-                    return 2 + countAdded;
-                }) :
-                countAdded;
-        });
-    });
-
-    var insertLongAds = memoize(function () {
-        return addArticleAds(8, getLongArticleRules()).then(function (countAdded) {
-            return 2 + countAdded;
-        });
-    });
-
-    function init() {
-        if (!commercialFeatures.articleBodyAdverts) {
-            return Promise.resolve(false);
-        }
-
-        var rules = getRules();
-
-        boot();
-
-        if (config.page.hasInlineMerchandise) {
-            addInlineMerchAd(getInlineMerchRules());
-        }
-
-        return addArticleAds(2, rules).then(function (countAdded) {
-            if (config.page.hasInlineMerchandise && countAdded === 0) {
-                waitForMerch().then(addSlots);
-            } else if (countAdded === 2) {
-                insertLongAds().then(addSlots);
-            }
         });
     }
 
-    return {
-        init: init,
-
-        '@@tests': {
-            waitForMerch: waitForMerch,
-            insertLongAds: insertLongAds
-        }
-    };
+    function addSlots(totalCount) {
```

Comment:
oh yeah, that. I removed it because it's one of those micro-optimisation I used to do which has no impact on the result, only it makes the code less readable. An improvement would be to remove the selector altogether and use an array of slots instead (since we create them in the module). I did that in the `-wide` version, in hindsight I should have done so here as well 😅